props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+    Vec3 diff = fusor.clash_mss();
+    float v;
+    if (clear) {
+      accel_ = accel;
+      diff = Vec3(0, 0, 0);
+      v = 0.0;
+    } else {
+  #ifndef PROFFIEOS_DONT_USE_GYRO_FOR_CLASH
+      v = (diff.len() + fusor.gyro_clash_value()) / 2.0;
+  #else
+      v = diff.len();
+  #endif
+    }
+    // If we're spinning the saber or if loud sounds are playing, 
+    // require a stronger acceleration to activate the clash.
+    if (v > (CLASH_THRESHOLD_G * (1
+          + fusor.gyro().len() / 500.0
+#if defined(ENABLE_AUDIO) && defined(AUDIO_CLASH_SUPPRESSION_LEVEL)
+          + dynamic_mixer.audio_volume() * (AUDIO_CLASH_SUPPRESSION_LEVEL * 1E-10) * dynamic_mixer.get_volume()
+#endif
+         ))) {
+      if ( (accel_ - fusor.down()).len2() > (accel - fusor.down()).len2() ) {
+        diff = -diff;
+      }
+
+      bool stab = (fabs(diff.x) > 2.0 * sqrtf(diff.y * diff.y + diff.z * diff.z)) &&
+            fusor.swing_speed() < 150;
+
+      if (!clash_pending1_) {
+        forward_stab_ = (stab && (diff.x < 0));
+        if (stab) PVLOG_DEBUG << "**** DoAccel >>>>>>>> - forward_stab_ = " << (forward_stab_ = (diff.x < 0)) << "\n";
+        clash_pending1_ = true;
+        pending_clash_is_stab1_ = stab;
+        pending_clash_strength1_ = v;
+      } else {
+        pending_clash_strength1_ = std::max<float>(v, (float)pending_clash_strength1_);
+      }
+    }
+    accel_ = accel;
+  }
+
+  // Handle lockup or drag for the main blade when only the main blade is on
+  void MainBladeLockupOrDrag() {
+    if (fusor.angle1() < -M_PI / 4) {
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_MAIN_BLADE_SET);
+    } else if (!battle_mode_) {
+      SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL, BC_MAIN_BLADE_SET);
+    }
+  }
+
+  // Handle lockup or drag for the second blade when only the second blade is on
+  void SecondBladeLockupOrDrag() {
+    if (fusor.angle1() > M_PI / 4) {
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_SECOND_BLADE_SET);
+    } else if (!battle_mode_) {
+     SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL, BC_SECOND_BLADE_SET);
+    }
+  }
+
+  // Handle lockup or drag for dual blades when both blades are on
+  void DualBladesLockupOrDrag() {
+    if (fusor.angle1() < -M_PI / 4) {
+      // Main blade is pointing down, and within drag range
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_MAIN_BLADE_SET);
+    } else if (fusor.angle1() > M_PI / 4) {
+      // Main blade is pointing up, and within drag range for the second blade
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_SECOND_BLADE_SET);
+    } else {
+      // Otherwise, apply normal lockup based on which blade is above the horizon
+      SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL, GetBladeAboveHorizon());
+    }
+  }
+
+#endif  // BC_DUAL_BLADES
+
+  RefPtr<BufferedWavPlayer> wav_player;
+
+  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+#ifdef BC_DUAL_BLADES
+    GetThrustBladeLocation();
```

Comment:
Why calculate this for every event and store it in a variable when it only seems to be used for *one* event?

Reply:
fixed.

Reply:
no wait, not yet.
It's not one event. there's all blades off, then if one is on.
almost fixed.